### PR TITLE
Use lodash's unionWith for union operator

### PR DESCRIPTION
**Pull request by alexkolson**
This should fix part of what is brought up in Picolab/pico-engine#23.

### CODE REVIEW

**Comment by alexkolson** [comment]
BTW, feel free to ignore this as you have probably already solved it / are going to solve it in a way filled with more awesomeness. Just saw this and thought I'd give it a shot.

File: src/index.js [hunk 1]
```diff
@@ -330,7 +330,11 @@ stdlib.values = function(val, path){
   return _.values(val);
 };
 stdlib.intersection = _.intersection;
-stdlib.union = _.union;
+stdlib.union = function() {
+  var args = Array.prototype.slice.call(arguments);
+  args.push(_.isEqual);
+  return _.unionWith.apply(_, args);
```
Comment (alexkolson) [inline src/index.js:336]: - Get the function's arguments in an array.
- Add the comparator function `_.isEqual` to the end of the arguments list for `_.unionWith`.
- Call `_.unionWith`, passing in the modified arguments with the comparator function.

**Comment by alexkolson** [comment]
Also kind of a side question but will there eventually be support for using NodeJS-supported ES6 syntactic sugar like arrow funcs, rest parameters, object shorthand, etc.?

**Comment by b1conrad** [comment]
Thanks, Alex, for jumping in! 

As to your question, do you mean support for these things within the new KRL language?

@farskipper will be fixing all of the set operator implementations, so I'll leave this to him. An alternative that I'm currently using to push forward my testing of KRL is this line:

`stdlib.union = _.partialRight(_.unionWith,_.isEqual);`

which I think is equivalent, and is working as expected

**Comment by windley** [comment]
Thanks for the help @alexkolson 

@b1conrad I'm assuming you'll manage pull requests here.

**Comment by farskipper** [comment]
Hey, thanks for helping, and feel free to do this to more of the set operators.

Can you add a test case that specifically asserts that union is doing _.isEquals equality rather than ===?

To run tests, just do `npm start` on your local. It'll automatically re-run tests as you make changes.

Thanks!

**Comment by alexkolson** [comment]
@farskipper I appreciate you even taking a look at this! Sounds like a plan.
@b1conrad That seems like an even more elegant way than mine to solve the issue. I would be inclined to switch the canonical stdlib union implementation to yours.
@windley Ha thanks. :)

Again, I just want to make clear that I don't wan't to step on anyone's toes/plans/implementations with the occasional PR and will always defer to @farskipper / @b1conrad for implementation details. 

I just still have a huge interest in seeing Kynetx/KRL/Pico ideology succeed, and adding a feature here and there or fixing a bug here and there is one of the ways I might be able to help from across the pond. 🇩🇪

**Comment by b1conrad** [comment]
Thank you, @alexkolson . We are grateful for your contribution and enthusiasm, and would love your active participation as your time permits. You are welcome here

There are no plans for any immediate activity on krl-stdlib, as @farskipper is concentrating on krl-compiler and pico-engine-core, and I am concentrating on pico-engine. So, please feel free to continue on this branch, and you won't be stepping on any toes. 

We do have a discipline of always having tests that validate the change, so if you would be so kind as to add a test or two, we'll merge your PR, and hope you'll want to work on the other set operators as well

I haven't forgotten your question, but am not equipped to answer yet. Would you be so kind as to add it as an issue to https://github.com/Picolab/pico-engine/issues

**Comment by b1conrad** [comment]
Thank you, @alexkolson !

**Comment by alexkolson** [comment]
Thanks @b1conrad for adding a test! Didn't get to it yet (obviously) 😉

**Pull request merged**